### PR TITLE
refactor: convert organizations command to yargs

### PR DESCRIPTION
**Pull request by rossiam**
* convert `organizations` command to yargs
* add unit tests for new and updated code
* added unit test for `chooseOrganizationFn` which had been left out in prior commit

### CODE REVIEW

File: src/commands/organizations.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+		.example([
+			['$0 organizations', 'list all organizations the user is a member of'],
+			[
+				'$0 organizations 1',
+				'display details for the first organization in the list retrieved by running ' +
+					'"smartthings organizations"',
+			],
+			[
+				'$0 organizations MyCompanyOrgName',
+				'display details for an organizations by name',
```
Comment (Sitlintac) [inline src/commands/organizations.ts:42]: Typo here: organization should be singluar

File: src/__tests__/lib/command/util/organizations-util.test.ts [hunk 2]
```diff
@@ -2,8 +2,17 @@ import { jest } from '@jest/globals'
 
 import type {
 	OrganizationResponse,
+	OrganizationsEndpoint,
+	SmartThingsClient,
 } from '@smartthings/core-sdk'
 
+import {
+	ValueTableFieldDefinition,
```
Comment (Sitlintac) [inline src/__tests__/lib/command/util/organizations-util.test.ts:10]: Does this want to be identified as a `type`?

File: src/__tests__/lib/command/util/organizations-util.test.ts [hunk 3]
```diff
@@ -2,8 +2,17 @@ import { jest } from '@jest/globals'
...
+} from '../../../../lib/table-generator.js'
+import {
+	createChooseFn,
+	type ChooseFunction,
+} from '../../../../lib/command/util/util-util.js'
 import {
 	InputDefinition,
```
Comment (Sitlintac) [inline src/__tests__/lib/command/util/organizations-util.test.ts:17]: As long as this file's being updated, does `InputDefinition` want to be identified as a `type` as well?

File: src/__tests__/commands/organizations.test.ts [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+import type { ArgumentsCamelCase, Argv } from 'yargs'
+
+import type {
+	OrganizationResponse,
+	OrganizationsEndpoint,
+	SmartThingsClient,
+} from '@smartthings/core-sdk'
+
+import type { APICommand, APICommandFlags } from '../../lib/command/api-command.js'
+import type { outputItemOrList, outputItemOrListBuilder } from '../../lib/command/listing-io.js'
```
Comment (Sitlintac) [inline src/__tests__/commands/organizations.test.ts:12]: These aren't types, are they?

**Review by Sitlintac — APPROVED** [review]

File: src/commands/organizations.ts (hunk 1 above)
Comment (rossiam) [reply src/commands/organizations.ts]: Heh, oops. Fixed.

File: src/__tests__/lib/command/util/organizations-util.test.ts (hunk 2 above)
Comment (rossiam) [reply src/__tests__/lib/command/util/organizations-util.test.ts:10]: Yup, fixed.

File: src/__tests__/lib/command/util/organizations-util.test.ts (hunk 3 above)
Comment (rossiam) [reply src/__tests__/lib/command/util/organizations-util.test.ts:17]: Yes, yes it does! 😄

Also, so do `selectDef`, `staticDef`,`undefinedDef`, and `createChooseFn`.

Good catch and fixed.

File: src/__tests__/commands/organizations.test.ts (hunk 4 above)
Comment (rossiam) [reply src/__tests__/commands/organizations.test.ts:12]: They are! And so much more. But alas, here I only care about the types.

**Pull request merged**